### PR TITLE
[REM][REF] stock: remove/clean code

### PR DESCRIPTION
**Pull request by ryv-odoo**
[REF] stock: move line: some guidelines

- move all the constrains at the same place, make sure they are private
- move all the onchanges at the same place, make sure they are private
- move `_get_similar_move_lines` helper at the bottom of the file with
the other helpers

[REF] stock: clean imports

Clean imports of the main stock module:
- Remove unsued import
- Follow odoo guidelines
- Sort them

[REM] stock: remove dead-code

Remove dead code of the main stock module:
- Unused actions
- Unused private methods
- Unused helper methods test

### CODE REVIEW

**Comment by ryv-odoo** [comment]
@Whenrow

File: addons/stock/models/stock_picking.py [hunk 1]
```diff
@@ -1293,4 +1293,3 @@ def _attach_sign(self):
             body=message,
         )
         return True
-
```
Comment (sle-odoo) [inline addons/stock/models/stock_picking.py]: don't remove this newline

File: addons/stock/views/report_stock_traceability.xml [hunk 2]
```diff
@@ -38,7 +38,7 @@
                             </span>
                         </t>
                         <t t-else="">
-                                <t t-raw="c"/>
+                            <t t-raw="c"/>
```
Comment (sle-odoo) [inline addons/stock/views/report_stock_traceability.xml]: useless

File: addons/stock/wizard/stock_backorder_confirmation.py [hunk 3]
```diff
@@ -68,4 +68,3 @@ def process_cancel_backorder(self):
                 .with_context(skip_backorder=True, picking_ids_not_to_backorder=self.pick_ids.ids)\
                 .button_validate()
         return True
-
```
Comment (sle-odoo) [inline addons/stock/wizard/stock_backorder_confirmation.py]: keep this newline

File: addons/stock/wizard/stock_immediate_transfer.py [hunk 4]
```diff
@@ -61,4 +61,3 @@ def process(self):
             pickings_to_validate = pickings_to_validate - pickings_not_to_do
             return pickings_to_validate.with_context(skip_immediate=True).button_validate()
         return True
-
```
Comment (sle-odoo) [inline addons/stock/wizard/stock_immediate_transfer.py]: keep this newline

File: addons/stock/wizard/stock_rules_report.py [hunk 5]
```diff
@@ -47,4 +47,3 @@ def print_report(self):
         self.ensure_one()
         data = self._prepare_report_data()
         return self.env.ref('stock.action_report_stock_rule').report_action(None, data=data)
-
```
Comment (sle-odoo) [inline addons/stock/wizard/stock_rules_report.py]: keep this newline

**Comment by sle-odoo** [comment]
robodoo r+ rebase-ff

**Pull request closed without merging**